unity-renderer/Assets/UIComponents/Scripts/Components/Button/ButtonComponentView.cs

```diff
@@ -60,30 +60,28 @@ public override void RefreshControl()
         SetIcon(model.icon);
     }
 
-    public bool IsInteractable() =>
-        button.interactable;
+    public bool IsInteractable() => button.interactable;
 
-    public void SetInteractable(bool isActive) =>
-        button.interactable = isActive;
+    public void SetInteractable(bool isActive) => button.interactable = isActive;
 
     public void SetText(string newText)
     {
         model.text = newText;
 
-        if (text == null)
-            return;
-
-        text.text = newText;
+        if (text)
+        {
+            text.text = newText;
+        }
     }
 
     public void SetIcon(Sprite newIcon)
     {
         model.icon = newIcon;
 
-        if (icon == null)
-            return;
-
-        icon.gameObject.SetActive(newIcon != null);
-        icon.sprite = newIcon;
+        if (icon)
+        {
+            icon.gameObject.SetActive(newIcon);
```

Comment:
IMO `icon.gameObject.SetActive(newIcon != null);` is easier to understand

Reply:
Maybe, but that can cause an unexpected behavior in some cases when using the `==` operator to check for `null` on `Unity.Object` derived classes, since it's overloaded by Unity, with extra checks (thus less performant) and unexpected behavior in some cases:
```
IEnumerator Demo()
    {
        GameObject go = new GameObject("Test Object");
        Destroy(go);

        yield return null;

        object objRef = go;
        Debug.Log($"objRef == null: {objRef == null}"); //prints False, unexpected.
        Debug.Log($"go == null: {go == null}"); //prints True, with overload, boxing and performance loss
        Debug.Log($"go == false: {go == false}"); //prints True, with no overload, boxing or performance loss
    }
```